reorganize show record metadata into sections

drh-stanford opened the pull request:
This PR fixes #1414 and is connected to #1441. It uses a section-based configuration (see `config.show.sections` which keeps all the configuration in ArticleController) and rewrites the `_show_default` partial.

Also upgrades to EDS 0.1.9pre for a dependency on multivalued authors.

Note there's no unit tests (pending EDS fixtures). It's a bit difficult to test this manually as finding example records in the wild isn't particularly effective.

### Typical record view

![screen shot 2017-07-07 at 9 51 32 am](https://user-images.githubusercontent.com/1861171/27968104-81d455c2-62fa-11e7-9b1d-ac66aab54401.png)

### Record view with missing Subjects section (as expected due to insufficient metadata)

![screen shot 2017-07-07 at 9 51 59 am](https://user-images.githubusercontent.com/1861171/27968100-7ea9d4bc-62fa-11e7-97a1-90ea4b76c16c.png)

jkeck commented on app/views/article/record/_section_fields.html.erb:
```diff
@@ -0,0 +1,14 @@
+ -%>
+  <% if value.present? %>
+    <dt><%= label -%></dt>
+    <dd><%= safe_join(Array.wrap(value)) -%></dd>
```
This is already being `safe_joined` in the presenter I think?  Do we need it both places?

jkeck commented on app/controllers/article_controller.rb:
```diff
@@ -70,30 +70,43 @@ def eds_init
+        eds_document_oclc: 'OCLC',
+        eds_document_type: 'Document Type',
+        eds_other_titles: 'Other Titles',
+        eds_physical_description: 'Physical Description'
```
One general concern I have about this approach is that we no longer have the ability to use configurations that we might want/need down the road (e.g. the `helper_method` that was removed, linking to searches, etc).

It's tough to say now if we'll need to break completely out of the BL configurations in order to derive our customized display, but as-is, this might make it harder to do something along the lines of link authors or subjects to searches (unless I'm missing something about this approach that would allow that).

One alternate approach might be to still configure all the show fields via `add_show_field` (allowing us to configure `heper_methods`, etc) but loop through a sections configuration in the UI can call `doc_presenter.field_value('field_name')` to continue to take advantage of the BL rendering pipeline.  This will do things like handle the `render_multivalued` that we've added to the presenter, and we can configure a helper to deal w/ the fields that we know to contain html (which is currently being handled by the `render_text_from_html` presenter method).

Happy to chat more about potential alternate approaches.

drh-stanford replied:
Thanks for the feedback. I think I've got something working now that keeps the `add_show_field` and feeds through the default presenter. The only concern I have is that rather than multiple values showing up on different lines, they are "and"ed together like so:

![screen shot 2017-07-06 at 2 56 02 pm](https://user-images.githubusercontent.com/1861171/27934593-4481c2c6-625b-11e7-8339-041b64ba4db4.png)

maybe there's a way to fix that in the rendering pipeline? Previously I was using the presenter to do a join with `'</dd><dd>'`...

jvine commented:
@drh-stanford now that I see more metadata - the Composed title is preferable to the Journal - can we switch their positions?

drh-stanford commented:
@jkeck I just did one last bit which was to add line breaks to the Authors and Subjects. I updated the screenshots and I removed the WIP -- it's ready to ship.
